source/boundary_temperature/interface.cc

```diff
@@ -215,39 +162,93 @@ namespace aspect
-        temperature = std::max(temperature,
-                               boundary_temperature_objects[i]->maximal_temperature(fixed_boundary_ids));
+      std::pair<std::multimap<types::boundary_id,unsigned int>::const_iterator,
+          std::multimap<types::boundary_id,unsigned int>::const_iterator> relevant_plugins =
+            boundary_temperature_map.equal_range(boundary_indicator);
+
+      for (; relevant_plugins.first != relevant_plugins.second; ++relevant_plugins.first)
+        {
+          const unsigned int plugin_index = relevant_plugins.first->second;
+          temperature = boundary_temperature_operators[plugin_index](temperature,
+                                                                     boundary_temperature_objects[plugin_index]->boundary_temperature(boundary_indicator,
+                                                                         position));
+        }
 
       return temperature;
     }
 
 
 
-    template <int dim>
-    const std::vector<std::string> &
-    Manager<dim>::get_active_boundary_temperature_names () const
+    namespace
     {
-      return model_names;
+      std_cxx11::tuple
+      <void *,
+      void *,
+      aspect::internal::Plugins::PluginList<Interface<2> >,
+      aspect::internal::Plugins::PluginList<Interface<3> > > registered_plugins;
     }
 
 
+
     template <int dim>
-    const std::vector<std_cxx11::shared_ptr<Interface<dim> > > &
-    Manager<dim>::get_active_boundary_temperature_conditions () const
+    void
+    Manager<dim>::register_boundary_temperature (const std::string &name,
+                                                 const std::string &description,
+                                                 void (*declare_parameters_function) (ParameterHandler &),
+                                                 Interface<dim> *(*factory_function) ())
     {
-      return boundary_temperature_objects;
+      std_cxx11::get<dim>(registered_plugins).register_plugin (name,
+                                                               description,
+                                                               declare_parameters_function,
+                                                               factory_function);
     }
 
 
+
     template <int dim>
     void
     Manager<dim>::declare_parameters (ParameterHandler &prm)
     {
+      prm.enter_subsection ("Model settings");
+      {
+        prm.declare_entry ("Fixed temperature boundary indicators", "",
+                           Patterns::List (Patterns::Anything()),
+                           "A comma separated list of names denoting those boundaries "
+                           "on which the temperature is fixed and described by the "
+                           "boundary temperature objects selected in the "
+                           "<Boundary temperature model/List of model names> parameter "
+                           "of this input file. All boundary indicators used by the geometry "
```

Comment:
Update: I see, you need it to apply the operators. So do we have plans for the future to allow several plugins for the velocity boundary conditions too? Because then I guess it would make sense to think about a format that would work for all three (temperature, composition, velocity).